client/src/Seed/Seed.js

```diff
@@ -81,28 +87,33 @@ class Seed extends Component {
   }
 
   iframeLoaded() {
-    this.channel = Channel.build({
-      window: this.iframe.contentWindow,
-      origin: this.props.origin,
-      scope: this.props.scope,
-      debugOutput: this.props.debugOutput,
-      onReady: () => {
-        this.ready = true;
-        this.props.onReady && this.props.onReady();
-      }
-    });
+    try {
+      this.channel = Channel.build({
+        window: this.iframe.contentWindow,
+        origin: this.props.origin,
+        scope: this.props.scope,
+        debugOutput: this.props.debugOutput,
+        onReady: () => {
+          this.ready = true;
```

Comment:
Multiple reasons:

1. Again, a state change triggers a re-rendering, which I do not want. The component *has already been rendered* and it's initialising. When the network thing is ready, the DOM should not change -> I do not want to render it again. 

2. The state is meant to be serialisable. If I serialise the state once the component is ready, I'd save `ready:true`. If I would re initialise it from the saved serialised state, it would set `ready:true` which is not false - the component would still need to talk with the iframe, create the channel and so on - it would lead to an inconsistent behaviour.

3. Do not confuse UI state with application state. UI State should be in React state - application state should probably be somewhere else. Very often these two states matches, but this is not the case. React State is a subset of State that, if changed, causes a React component to re-render itself.

4. [If it's not used in rendering, it should not be in the state](https://twitter.com/dan_abramov/status/749710501916139520?lang=en)

Does that clarify?